IO_rot.f90

```diff
@@ -348,101 +364,122 @@ subroutine mk_config
     integer ind_e
     open(11,file='setting.txt',form='formatted',status='old')
     open(99,file=trim(outdir) // "/config.txt",status="replace",form="formatted")
-    do 
+    do
        read(11,"(A)",end=999)tmp
-       call get_word(tmp,key,ind_e)       
+       call get_word(tmp,key,ind_e)
        if(ind_e.gt.1) write(99,"(A)")key//":"//tmp(1:ind_e-1)
     enddo
 999 continue
     close(11)
     write(99,"(A)")"ENDSETTING"
     write(99,*)flag_mhd,flag_pip, " #mhd and pip flag"
     write(99,*)nvar_h,nvar_m, " #number of variables"
-    write(99,*)ix,jx,kx, " # used grid numbers" 
-    write(99,*)margin, " # used margin grid numbers" 
+    write(99,*)ix,jx,kx, " # used grid numbers"
+    write(99,*)margin, " # used margin grid numbers"
     write(99,*)gm," #Abiabatic constant"
     write(99,*)flag_bnd
 !    write(99,*)flag_damp,damp_time, "velocity damping"
     if(flag_mpi.eq.1) then
        write(99,*)mpi_siz," #mpi domain size"
     endif
-    close(99)    
+    close(99)
   end subroutine mk_config
-  
+
   !! restart routine should be modified
   subroutine restart
     integer tmp,out_tmp
     character*200 line
     character*15 key
-!    open(99,file=trim(indir)//"result.txt",status="old",form="formatted"
 
     !Modification restart setting 2015/08/27 NN======================
-
     open(restart_unit,file=trim(indir)//"config.txt",status="old",form="formatted")
-    do  
-       read(restart_unit,*,end=111)line
-       if(trim(line)=="ENDSETTING") exit
+    do
+      read(restart_unit,*,end=111)line
+      if(trim(line)=="ENDSETTING") exit
     enddo
 111 continue
     read(restart_unit,*)flag_mhd,flag_pip
     read(restart_unit,*)nvar_h,nvar_m
-    read(restart_unit,*)ix,jx,kx
+    read(restart_unit,*)
```

Comment:
Removing `ix,jx,kx` and `mpi_siz` here allow a restarted simulation to use a different number of mpi ranks.